tsdb/shard.go

```diff
@@ -544,45 +528,24 @@ func (s *Shard) WritePointsWithContext(ctx context.Context, points []models.Poin
 		return err
 	}
 
-	// see if our engine is capable of WritePointsWithContext
-	type contextWriter interface {
-		WritePointsWithContext(context.Context, []models.Point) error
-	}
-	switch eng := engine.(type) {
-	case contextWriter:
-		if err := eng.WritePointsWithContext(ctx, points); err != nil {
-			atomic.AddInt64(&s.stats.WritePointsErr, int64(len(points)))
-			atomic.AddInt64(&s.stats.WriteReqErr, 1)
-			return fmt.Errorf("engine: %s", err)
-		}
-	default:
-		// Write to the engine.
-		if err := engine.WritePoints(points); err != nil {
-			atomic.AddInt64(&s.stats.WritePointsErr, int64(len(points)))
-			atomic.AddInt64(&s.stats.WriteReqErr, 1)
-			return fmt.Errorf("engine: %s", err)
+	engineTracker := func (points, values int64){
```

Comment:
This will likely cause an allocation per shard per write request - not too worried about that performance wise, thoughts from @dgnorton ?